securesystemslib/metadata.py

```diff
@@ -67,15 +66,13 @@ def from_dict(cls, data: dict) -> "Envelope":
 
         signatures = []
         for signature in data["signatures"]:
-            if formats.GPG_SIGNATURE_SCHEMA.matches(signature):
-                raise NotImplementedError
+            formats.ANY_SIGNATURE_SCHEMA.check_match(signature)
 
-            if formats.SIGNATURE_SCHEMA.matches(signature):
+            try:
+                signatures.append(GPGSignature.from_dict(signature))
+            except KeyError:
```

Comment:
I'm unsure if I wouldn't like it better to [dispatch based on the distinguishing feature](https://github.com/theupdateframework/python-tuf/blob/e07fa2776283ebafc8fbab4cd386aee0a971fc13/tuf/api/metadata.py#L172-L184), i.e. `other_headers`.

Your code makes for better decoupling but has worse performance and is less explicit. What do you think, @PradyumnaKrishna?

Reply:
I moved back to `formats.SCHEMA.matches`.